### PR TITLE
PRO-3188 env var to log all middleware and routes touched by a URL

### PR DESCRIPTION
**Pull request by boutell**
## Summary

env var to log all middleware and routes touched by a URL.

Same general pattern as the existing `APOS_LOG_ALL_QUERIES` and `APOS_TRACE_STARTUP`.

Needs a corresponding docs PR.

## What are the specific steps to test this change?

```
APOS_LOG_ALL_ROUTES=1 node app
```

Now access something.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Review by ETLaurent — CHANGES_REQUESTED** [review]
I've got the following error when testing it on testbed:

```
/ invokes middleware found at @apostrophecms/asset:serveStaticAssets
/ invokes middleware found at @apostrophecms/express:attachUtilityMethods
/ invokes middleware found at @apostrophecms/express:createDataAndGuards
/ invokes middleware found at @apostrophecms/express:session
/ invokes middleware found at @apostrophecms/express:cookieParser
/ invokes middleware found at @apostrophecms/express:apiKeys
/ invokes middleware found at @apostrophecms/express:
/ invokes middleware found at @apostrophecms/express:bearerTokens
/ invokes middleware found at @apostrophecms/express:csrf
/ invokes middleware found at @apostrophecms/express:urlencodedParser
/ invokes middleware found at @apostrophecms/express:jsonParser
/ invokes middleware found at @apostrophecms/login:passportInitialize
/ invokes middleware found at @apostrophecms/login:passportExtendLogin
/ invokes middleware found at @apostrophecms/login:passportSession
/ invokes middleware found at @apostrophecms/login:removeUserForDraftSharing
/ invokes middleware found at @apostrophecms/login:honorLoginInvalidBefore
/ invokes middleware found at @apostrophecms/login:addUserToData
/ invokes middleware found at @apostrophecms/login:addLoggedInCookie
/ invokes middleware found at @apostrophecms/i18n:acceptCrossDomainSessionToken
/ invokes middleware found at @apostrophecms/i18n:locale
TypeError: trust is not a function
    at IncomingMessage.protocol (/Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/request.js:312:8)
    at Object.setPrefixUrls (/Users/etienne/Sites/Apostrophe/apostrophe/modules/@apostrophecms/i18n/index.js:598:40)
    at locale (/Users/etienne/Sites/Apostrophe/apostrophe/modules/@apostrophecms/i18n/index.js:257:14)
    at /Users/etienne/Sites/Apostrophe/apostrophe/modules/@apostrophecms/express/index.js:407:20
    at /Users/etienne/Sites/Apostrophe/testbed/node_modules/@opentelemetry/instrumentation-express/build/src/instrumentation.js:160:37
    at Layer.handle [as handle_request] (/Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/index.js:328:13)
    at /Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/index.js:346:12)
    at next (/Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/index.js:280:10)
    at acceptCrossDomainSessionToken (/Users/etienne/Sites/Apostrophe/apostrophe/modules/@apostrophecms/i18n/index.js:171:18)
    at /Users/etienne/Sites/Apostrophe/apostrophe/modules/@apostrophecms/express/index.js:407:20
    at /Users/etienne/Sites/Apostrophe/testbed/node_modules/@opentelemetry/instrumentation-express/build/src/instrumentation.js:160:37
    at Layer.handle [as handle_request] (/Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/index.js:328:13)
    at /Users/etienne/Sites/Apostrophe/apostrophe/node_modules/express/lib/router/index.js:286:9
```

**Comment by boutell** [comment]
Marking this draft until this morning's release because it didn't make the window for PRs.

**Comment by boutell** [comment]
@ETLaurent ready for re-review

**Review by ETLaurent — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## UNRELEASED
 
+### Adds
+
+* New `APOS_LOG_ALL_ROUTES` environment variable. If set, Apostrophe logs information about all middleware functions and routes that are executed on behalf of a particular URL.
```
Comment (ETLaurent) [inline CHANGELOG.md:7]:
```suggestion
* New `APOS_LOG_ALL_ROUTES` environment variable. If set, Apostrophe logs information about all middleware functions and routes that are executed on behalf of a particular URL.

### Fixes

```

(not related to this PR)

**Comment by boutell** [comment]
Accepting the reviewer's suggestion discards their review? Boo. Here it is again

**Review by ETLaurent — APPROVED** [review]

**Pull request merged**